src/Classes/Entities/EventEntity.php

```diff
@@ -28,5 +28,173 @@ public function __construct(
         $this->startTime = ($this->startTime ?? $startTime);
         $this->endTime = ($this->endTime ?? $endTime);
         $this->notes = ($this->notes ?? $notes);
+
+        $this->sanitiseData();
+    }
+
+    /**
+     * Will sanitise all the fields for an event.
+     */
+    private function sanitiseData()
+    {
+        $this->name = $this->sanitiseString($this->name);
+        $this->name = self::validateLength($this->name, 255);
+        $this->category = (int)$this->category;
```

Comment:
Would it be possible to check if the category exists in the database?

Reply:
yes but not in a clean way